test/library.cpp

```diff
@@ -486,6 +486,14 @@ TEST_F(LibraryTest, filterByName)
   EXPECT_FILTER_RESULTS(kiwix::Filter().name("wikibooks_de"),
     "An example ZIM archive"
   );
+
+  EXPECT_FILTER_RESULTS(kiwix::Filter().query("name:wikibooks_de"),
+    "An example ZIM archive"
+  );
```

Comment:
How is it working with `kiwix::Filter().name("foo").query("name:bar")` ?

Reply:
I don't think we have to define the behaviour for this case (rather we can define the result to be unspecified).

Reply:
Ok, we will see what to do if somehow we have this case. But as you, I doubt we have to.